Core/Core/UIViews/CoreDatePicker.swift

```diff
@@ -0,0 +1,73 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2022-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public class CoreDatePicker: UIDatePicker {
+    public let toolbar = UIToolbar()
+    public weak var datePickerDelegate: DatePickerProtocol?
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.frame.size.width = frame.width
+        preferredDatePickerStyle = .wheels
+        datePickerMode = .dateAndTime
+        datepickerSetup()
+        toolbarSetup()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    func toolbarSetup() {
```

Comment:
Make sure to apply to correct visibility modifiers everywhere. If we only use this function internally make it private.